internal/crypto/src/cose/certificate_acceptance_policy.rs

```diff
@@ -0,0 +1,260 @@
+// Copyright 2023 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License,
+// Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+// or the MIT license (http://opensource.org/licenses/MIT),
+// at your option.
+
+// Unless required by applicable law or agreed to in writing,
+// this software is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR REPRESENTATIONS OF ANY KIND, either express or
+// implied. See the LICENSE-MIT and LICENSE-APACHE files for the
+// specific language governing permissions and limitations under
+// each license.
+
+#![allow(clippy::doc_lazy_continuation)] // Clippy and rustfmt aren't agreeing at the moment. :-(
+
+use std::{collections::HashSet, error::Error, fmt, str::FromStr};
+
+use asn1_rs::{oid, Oid};
+use x509_parser::{extensions::ExtendedKeyUsage, pem::Pem};
+
+/// A `CertificateAcceptancePolicy` retains information about trust anchors and
+/// allowed EKUs to be used when verifying C2PA signing certificates.
+#[derive(Debug)]
+pub struct CertificateAcceptancePolicy {
+    /// Trust anchors (root X.509 certificates) in DER format.
+    trust_anchor_ders: Vec<Vec<u8>>,
+
+    /// End-entity certificaters (root X.509 certificates) in DER format.
+    end_entity_cert_ders: Vec<Vec<u8>>,
+
+    /// Additional extended key usage (EKU) OIDs.
+    additional_ekus: HashSet<String>,
+}
+
+impl Default for CertificateAcceptancePolicy {
+    fn default() -> Self {
+        let mut this = CertificateAcceptancePolicy {
+            trust_anchor_ders: vec![],
+            end_entity_cert_ders: vec![],
+            additional_ekus: HashSet::default(),
+        };
+
+        this.add_valid_ekus(include_bytes!("./valid_eku_oids.cfg"));
+
+        // In testing configs, also add debug/trust anchors.
+        #[cfg(test)]
```

Comment:
Will this work for unit tests in c2pa-rs.  We plan to turn on trust checks by default for all unit tests since trust is required for 2.x.

Reply:
The root certs that are added at line 49 are only the test certs. When we have the C2PA root bundle, we could add another `add_trust_anchors` call with those roots and not gate that call on `#[cfg(test)]`.

Reply:
@mauricefisher64 see https://github.com/contentauth/c2pa-rs/pull/779/files#diff-3867f533b55dbd88258138ed6ea7d7127ee605b3b68750c6b0174baa6efd76ebR222-R229 (just added)